### PR TITLE
Add support for spawning local wasm files

### PR DESCRIPTION
**Pull request by maminrayej**
This PR adds support for spawning local wasm files.

### CODE REVIEW

File: lib/wasix/src/bin_factory/exec.rs [hunk 1]
```diff
@@ -29,15 +29,25 @@ pub async fn spawn_exec(
...
+pub async fn spawn_exec_wasm(
+    wasm: &[u8],
+    name: &str,
+    env: WasiEnv,
+    runtime: &Arc<dyn Runtime + Send + Sync + 'static>,
+) -> Result<TaskJoinHandle, SpawnError> {
     // Load the module
     let module = spawn_load_module(&env, name, wasm, runtime).await?;
 
-    // Spawn union the file system
```
Comment (john-sharratt) [inline lib/wasix/src/bin_factory/exec.rs:38]: not sure what this comment means, does it mean Spawn the union file system? I don't think we spawn it though we set it up

File: lib/wasix/src/bin_factory/mod.rs [hunk 2]
```diff
@@ -145,6 +144,90 @@ impl BinFactory {
...
+                    spawn_exec_wasm(&bytes, name.as_str(), env, &self.runtime).await
+                }
+                Executable::BinaryPackage(pkg) => {
+                    spawn_exec(pkg, name.as_str(), store, env, &self.runtime).await
+                }
+            }
+        })
+    }
+
+    // TODO: remove allow once BinFactory is refactored
```
Comment (john-sharratt) [inline lib/wasix/src/bin_factory/mod.rs:179]: Can't we use the bin factory here now?

**Review by john-sharratt — APPROVED** [review]

File: lib/wasix/src/bin_factory/exec.rs (hunk 1 above)
Comment (maminrayej) [reply lib/wasix/src/bin_factory/exec.rs:38]: You're right. This is a leftover comment from `spawn_exec`.

File: lib/wasix/src/bin_factory/mod.rs (hunk 2 above)
Comment (maminrayej) [reply lib/wasix/src/bin_factory/mod.rs:179]: I'm not sure what you mean.

**Review by john-sharratt — APPROVED** [review]

**Pull request merged**